### ISSUE TITLE
segfault with SLSQP

### ISSUE THREAD

**Issue by eppravitra** [stevengj/nlopt#495]
I think these lines in slsqp.c are not quite correct.
```
	  case 1:{ /* don't need grad unless we don't have it yet */
	      double *newgrad = 0;
	      double *newcgrad = 0;
	      if (want_grad) {
		  newgrad = grad;
		  newcgrad = cgradtmp;
	      }
	      feasible_cur = 1; infeasibility_cur = 0;
	      fcur = f(n, xcur, newgrad, f_data);
```
I ran into a situation where want_grad somehow got set to 0, so newgrad never got assigned a value.
I got a segfault the next time it tries to evaluate the objective function.

**Comment by jschueller** [comment]
please provide a minimal reproducer

**Comment by eppravitra** [comment]
Hi,

I figured out what happened. It was my bad. There is nothing wrong with the SQP code.

In my objective function, I computed the gradient. I was trying to use that gradient value in the expression of the objective function.

I now figured that I shouldn't. That's why you guys have that
`if(!grad.empty())`
 thing in the example.

Thank you!!





For someone else who might run into this, just write the value of the gradient. You are not supposed to read it out.
